File: qiniulib/src/main/java/com/shuai/qiniulib/QiNiuConfig.java
package com.shuai.qiniulib;

import com.qiniu.util.Auth;

public class QiNiuConfig {

    private static QiNiuConfig.Config sConfig;

    public static Config getConfig() {
        return sConfig;
    }

    public static QiNiuConfig.Config init(String accessKey, String secretKey) {
        if (sConfig == null) {
            synchronized (QiNiuConfig.class) {
                if (sConfig == null) {
                    sConfig = new QiNiuConfig.Config(accessKey, secretKey);
                }
            }
        }
        return sConfig;
    }

    public static class Config {
        private final Auth auth;

        public Config(String accessKey, String secretKey) {
            this.auth = Auth.create(accessKey, secretKey);
        }

        public Auth getAuth() {
            return auth;
        }
    }
}

File: qiniulib/src/main/java/com/shuai/qiniulib/QiNiuConfig.kt
package com.shuai.qiniulib

import com.qiniu.util.Auth

class QiNiuConfig {

    companion object {

        var config: Config? = null
            private set

        @JvmStatic
        fun init(accessKey: String, secretKey: String): Config? {
            if (config == null) {
                synchronized(QiNiuConfig::class.java) {
                    if (config == null) {
                        config = Config(accessKey, secretKey)
                    }
                }
            }
            return config
        }
    }


    class Config(accessKey: String, secretKey: String) {
        val auth: Auth = Auth.create(accessKey, secretKey)

    }
}